#166114088 Assigner history

3N61N33R opened the pull request:
#### What does this PR do?
- adds assigner field to allocation history model

#### Description of Task to be completed?
Given an admin with access
When I request to view an asset
Then I should see the list of assignee history with timestamps and the list of people who assigned it in different instances

#### How should this be manually tested?


#### Any background context you want to provide?
Currently, we only have a list of the people assigned in the allocation history.

#### What are the relevant pivotal tracker stories?
[#166114088](https://www.pivotaltracker.com/story/show/166114088)

Commit:
feat(assigner history): track asset assigner

- adds assigner field to allocation history model

[Finishes #166114088]

Commit:
feat(assigner history): track asset assigner

- adds assigner field to allocation history model

[Finishes #166114088]

kmazi reviewed (changes requested):
It's nice to have integration tests at least for your feature.

Commit:
- adjust tests

Commit:
- adjust tests

kmazi commented on api/tests/test_allocation_api.py:
```diff
@@ -24,16 +24,24 @@ def test_non_authenticated_user_view_assets(self):
-        response = client.get(
+
+        data = {"asset": self.asset.id, "current_owner": self.asset_assignee.id}
+        client.post(
```
Is this test for posting allocations or getting allocations? I also see that the method name says get but the description says post and also the line above.
For test purpose you could just create an additional AllocationHistory object and save it in the database.

3N61N33R replied:
As written it is for `getting allocations` 
I added the post since the response had no allocations whatsoever and I couldn't assert that the response had the assigner
Hadn't looked at the docstring since I was adjusting the test, an oversight on my end
Will correct that, thanks

kmazi replied:
You can create the allocation history here with AllocationHistory.objects.create() rather than making a post.

kmazi commented on api/tests/test_allocation_api.py:
```diff
@@ -24,16 +24,24 @@ def test_non_authenticated_user_view_assets(self):
+        self.assertEqual(len(data), AllocationHistory.objects.count())
         self.assertEqual(response.status_code, 200)
+        self.assertEqual(data[0]["assigner"], self.other_user.email)
+        self.assertIn("assigner", data[0])
```
You already used the "assigner" key in the previous assertion which made this somewhat unnecessary.

kmazi reviewed (commented):
There seems to be no test that shows, as contained in the story description, that when a user views an asset then they should see a list of allocation history of both assigner and assignee.

3N61N33R commented:
> There seems to be no test that shows, as contained in the story description, that when a user views an asset then they should see a list of allocation history of both assigner and assignee.

This had already been tested for in a different PR on the assignee that had been merged
What is on this PR are the adjustments for the assigner

kmazi commented:
> > There seems to be no test that shows, as contained in the story description, that when a user views an asset then they should see a list of allocation history of both assigner and assignee.
> 
> This had already been tested for in a different PR on the assignee that had been merged
> What is on this PR are the adjustments for the assigner

Then let the tests show the adjustment you've made in the response when viewing details of an asset. Isn't that what's in the story? ```When I request to view an asset
Then I should see the list of assignee history with timestamps and the list of people who assigned it in different instances```

kmazi commented:
Your tests are more like a documentation of your implementation you know.

Commit:
- test assigner in asset view allocation history

Commit:
- test assigner in asset view allocation history

Commit:
- trim trailing whitespace

Commit:
- trim trailing whitespace

RonKbS committed:
#165377092 Allow posting and getting of asset invoice receipts (#304)

* feat(asset): allow posting and getting of asset invoice receipts

- allow only a super admin to post an asset's invoice_receipt
- allow only an admin to view an asset's invoice_receipt

[Finishes #165377092]

* - make restrict_to_only_super_users an internal function
- add assertions for file_upload

[Finishes #165377092]

*  - fix black-formatting error in test_manage_asset_api file

* - add settings to allow viewing of uploaded files
- add test that shows a super_admin can patch an asset's invoice receipt

* - remove test file uploaded during patching by super-admin

Commit:
Merge branch 'ft-assigner-history-166114088' of https://github.com/AndelaOSP/art-backend into ft-assigner-history-166114088

Commit:
fix migration conflict
i
